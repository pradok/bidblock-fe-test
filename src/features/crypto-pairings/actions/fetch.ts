import { actionTypes as at } from './constants';
import { CryptoPairingsCollection, CryptoPairSymbol } from '../models';

export const fetchCryptoPairings = (symbol: CryptoPairSymbol = 'ALL') => {
  return {
    symbol,
    type: at.CRYPTO_PAIRINGS_FETCH,
  };
};

export const fetchSuccess = (result: CryptoPairingsCollection) => {
  return {
    payload: result,
    type: at.CRYPTO_PAIRINGS_FETCH_SUCCESS,
  };
};

export const fetchError = (error: Error) => {
  return {
    payload: error,
    type: at.CRYPTO_PAIRINGS_FETCH_ERROR,
  };
};