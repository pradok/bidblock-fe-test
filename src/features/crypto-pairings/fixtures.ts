import {CryptoPairingsCollection} from "./models";

const pairings: CryptoPairingsCollection = [
  [
    "tBTCUSD", 6239.9, 34.92908368, 6240, 35.7623858, -30.1, -0.0048, 6239.9, 18771.42426519, 6339.9, 6125
  ],
  [
    "tLTCUSD", 75.999, 1099.6280602, 76, 486.87405918, -1.648, -0.0212, 76, 110391.68659301, 78.66, 75.049
  ]
];

export {pairings};