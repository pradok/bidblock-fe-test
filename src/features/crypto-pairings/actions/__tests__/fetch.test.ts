import {fetchCryptoPairings, fetchSuccess, fetchError} from '../fetch';
import {pairings} from '../../fixtures';

const error: Error = {
  name: 'Error Name',
  message: 'Error fetching collection',
};

describe('features -> product -> actions', () => {
  it('fetchProducts', () => {
    expect(fetchCryptoPairings()).toMatchSnapshot();
  });
  it('fetchProducts', () => {
    expect(fetchSuccess(pairings)).toMatchSnapshot();
  });
  it('fetchProducts', () => {
    expect(fetchError(error)).toMatchSnapshot();
  });
});
