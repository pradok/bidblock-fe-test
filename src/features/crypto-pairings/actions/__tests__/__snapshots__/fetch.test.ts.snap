// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features -> product -> actions fetchProducts 1`] = `
Object {
  "symbol": "ALL",
  "type": "CRYPTO/PAIRINGS_FETCH",
}
`;

exports[`features -> product -> actions fetchProducts 2`] = `
Object {
  "payload": Array [
    Array [
      "tBTCUSD",
      6239.9,
      34.92908368,
      6240,
      35.7623858,
      -30.1,
      -0.0048,
      6239.9,
      18771.42426519,
      6339.9,
      6125,
    ],
    Array [
      "tLTCUSD",
      75.999,
      1099.6280602,
      76,
      486.87405918,
      -1.648,
      -0.0212,
      76,
      110391.68659301,
      78.66,
      75.049,
    ],
  ],
  "type": "CRYPTO/PAIRINGS_FETCH_SUCCESS",
}
`;

exports[`features -> product -> actions fetchProducts 3`] = `
Object {
  "payload": Object {
    "message": "Error fetching collection",
    "name": "Error Name",
  },
  "type": "CRYPTO/PAIRINGS_FETCH_ERROR",
}
`;
